File: examples/benchmark/graph.ts
import { Annotation, END, START, StateGraph } from "@langchain/langgraph";

const BenchmarkState = Annotation.Root({
  seed: Annotation<number>,
  increment: Annotation<number>,
  intermediate: Annotation<number>,
  result: Annotation<number>,
});

/** Two deterministic 200 ms steps; { seed: 7, increment: 5 } produces result 24. */
export const benchmark = new StateGraph(BenchmarkState)
  .addNode("add", async state => {
    await Bun.sleep(200);
    return { intermediate: state.seed + state.increment };
  })
  .addNode("double", async state => {
    await Bun.sleep(200);
    return { result: state.intermediate * 2 };
  })
  .addEdge(START, "add")
  .addEdge("add", "double")
  .addEdge("double", END)
  .compile();

File: benchmarks/aegra/graph.py
"""Deterministic two-stage async graph for the Aegra/Valida benchmark."""

import asyncio
from typing import TypedDict
from langgraph.graph import END, START, StateGraph


class State(TypedDict, total=False):
    seed: int
    increment: int
    intermediate: int
    result: int


async def add(state: State) -> dict[str, int]:
    await asyncio.sleep(0.2)
    return {"intermediate": state["seed"] + state["increment"]}


async def double(state: State) -> dict[str, int]:
    await asyncio.sleep(0.2)
    return {"result": state["intermediate"] * 2}


builder = StateGraph(State)
builder.add_node("add", add)
builder.add_node("double", double)
builder.add_edge(START, "add")
builder.add_edge("add", "double")
builder.add_edge("double", END)
graph = builder.compile()
